app/src/main/java/space/taran/arknavigator/utils/FileUtils.kt

```diff
@@ -18,38 +33,132 @@ enum class Sorting {
+fun isVideo(filePath: Path): Boolean {
+    if (!isValidFile(filePath)) return false
+
+    val extension = extension(filePath)
+    return acceptedVideoExt.contains(extension)
+}
+
+fun isValidFile(filePath: Path): Boolean {
+    return getFileSizeMB(filePath) != 0
+}
+
+fun isPDF(filePath: Path): Boolean {
+    return extension(filePath) == ".pdf"
+}
+
+fun isFormat(filePath: Path, format: String): Boolean {
+    return extension(filePath) == format
+}
+
+fun getPdfPreviewsFolder(): File =
+    File("${App.instance.cacheDir}/$PDF_PREVIEW_FOLDER_NAME")
+
+fun getPdfPreviewByID(id: Long): Path {
+    val pathName = "${App.instance.cacheDir}/$PDF_PREVIEW_FOLDER_NAME/$id.png"
+    return File(pathName).toPath()
+}
+
+fun getSavedPdfPreviews(): List<Long?>? =
+    getPdfPreviewsFolder()
+        .listFiles()
+        ?.map {
+            it.nameWithoutExtension.toLongOrNull()
+        }
+
+fun getVideoInfo(filePath: Path): MutableMap<Preview.ExtraInfoTag, String> {
+    val retriever = MediaMetadataRetriever()
+
+    retriever.setDataSource(App.instance, Uri.fromFile(filePath.toFile()))
+    val timeMillis =
+        retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)?.toLong() ?: 0L
+    val width =
+        retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH)?.toLong() ?: 0L
+    val height =
+        retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT)?.toLong() ?: 0L
+
+
+    var minutes = TimeUnit.MILLISECONDS.toMinutes(timeMillis).toString()
+    val minutesPassedInMillis = TimeUnit.MINUTES.toMillis(minutes.toLong()).toString()
+    var seconds =
+        TimeUnit.MILLISECONDS.toSeconds(timeMillis - minutesPassedInMillis.toLong()).toString()
+
+    if (minutes.length == 1) minutes = "0$minutes"
+    if (seconds.length == 1) seconds = "0$seconds"
+
+    val duration = "$minutes:$seconds"
+    val resolution = convertToResolution(width, height)
+
+    val mutableMap = mutableMapOf<Preview.ExtraInfoTag, String>()
+    mutableMap[Preview.ExtraInfoTag.MEDIA_DURATION] = duration
+
+    if (resolution != null)
+        mutableMap[Preview.ExtraInfoTag.MEDIA_RESOLUTION] = resolution
+
+    retriever.release()
+    return mutableMap
+}
+
+fun convertToResolution(width: Long, height: Long): String? {
+    val resolutionPair = listOf(width, height)
+
+    return when {
+        resolutionPair.containsAll(listOf(256, 144)) -> "144p"
+        resolutionPair.containsAll(listOf(426, 240)) -> "240p"
+        resolutionPair.containsAll(listOf(640, 360)) -> "360p"
+        resolutionPair.containsAll(listOf(854, 480)) -> "480p"
+        resolutionPair.containsAll(listOf(1280, 720)) -> "720p"
+        resolutionPair.containsAll(listOf(1920, 1080)) -> "1080p"
+        resolutionPair.containsAll(listOf(2560, 1440)) -> "1440p"
+        resolutionPair.containsAll(listOf(3840, 2160)) -> "2160p"
+        else -> null
+    }
+}
+
+fun createPdfPreview(filePath: Path, context: Context? = null): Bitmap {
```

Comment:
Let's move this logic into dedicated place and not pollute `FileUtils` :)
This will also allow us to declare values like `pageNumber` as constants.